core/src/main/java/org/apache/carbondata/core/dictionary/generator/ServerDictionaryGenerator.java

```diff
@@ -52,7 +52,11 @@ public void initializeGeneratorForTable(DictionaryMessage key) {
             key.getTableUniqueName(), key.getColumnName());
     // initialize TableDictionaryGenerator first
     if (tableMap.get(key.getTableUniqueName()) == null) {
-      tableMap.put(key.getTableUniqueName(), new TableDictionaryGenerator(dimension));
+      synchronized (tableMap) {
+        if (tableMap.get(key.getTableUniqueName()) == null) {
+          tableMap.put(key.getTableUniqueName(), new TableDictionaryGenerator(dimension));
+        }
```

Comment:
here only more else is required.
 else {
        tableMap.get(key.getTableUniqueName()).updateGenerator(dimension);		        tableMap.get(key.getTableUniqueName()).updateGenerator(dimension);
      }